nomad/structs/structs.go

```diff
@@ -4995,6 +5013,13 @@ func (n *Namespace) Copy() *Namespace {
 	nc := new(Namespace)
 	*nc = *n
 	nc.Hash = make([]byte, len(n.Hash))
+	if n.Capabilities != nil {
+		c := new(NamespaceCapabilities)
+		*c = *n.Capabilities
+		copy(c.EnabledTaskDrivers, n.Capabilities.EnabledTaskDrivers)
+		copy(c.DisabledTaskDrivers, n.Capabilities.DisabledTaskDrivers)
```

Comment:
Sorry, I missed these two lines were supposed to have resolved my comment here https://github.com/hashicorp/nomad/pull/11813#discussion_r782433839 but it turns out this copies the pointers the strings point to, not the strings. Which means this can happen: https://play.golang.com/p/IbahWCEniLV

We have a helper function [`CopySliceString`](https://github.com/hashicorp/nomad/blob/v1.2.6/helper/funcs.go#L383-L392) for just this use case.

Reply:
Ups, I do know C so I should be better at this. Yet somehow Go is still confusing -- thank you for teaching me and will fix asap.